packages/cw721/src/msg.rs

```diff
@@ -189,7 +189,10 @@ pub enum Cw721QueryMsg<TMetadataExtension> {
     // -- "type annotations needed: cannot infer type for type parameter `TMetadataExtension` declared on the enum `Cw721QueryMsg`"
     /// Do not use - dummy extension query, needed for inferring type parameter during compile
     #[returns(())]
-    Extension { msg: TMetadataExtension },
+    Extension {
+        msg: TQueryExtensionMsg,
+        phantom: Option<TMetadataExtension>, // dummy field to infer type
+    },
```

Comment:
included the optional `phantom` field here to satisfy the compile error where all declared generics must be used in the enum. not sure if this is the way so please lmk if otherwise